### DISCUSSION TITLE
Enhancement to Chips Component (add input clear)

### DISCUSSION THREAD

**Discussion by blake-w** [primefaces/community#1009] [PrimeNG] [answered]
I am creating a discussion and a pull request to enhance the Chips component. Currently, there is no way to clear the input field. This is problematic when you display the Chips component in a dialog, then you hide the dialog, and upon reopening the dialog any unsubmitted Chip will still linger in the input field.

Ideally, this would be tied into the hide event as it would be my expected behavior. However, that does align with how Primeng usually handles hiding components (does not hard-clear/edit fields).

You are currently required to access the input field through the element query selector:
`chipsRef.el.nativeElement.querySelector("input");`

I believe this enhancement would prevent others from manually targeting and clearing the input field and it could be done with a public method.

```
public clearInput(): void {
    this.inputViewChild.nativeElement.value = '';
}
```

Pull request: https://github.com/primefaces/primeng/pull/14620

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks a lot for the effort and support, we'll review the PR in 17.5.0 or the next release.
